src/structures/User.js

```diff
@@ -45,11 +45,13 @@ class User extends Base {
       this.username = null;
     }
 
-    /**
-     * Whether or not the user is a bot
-     * @type {boolean}
-     */
-    this.bot = Boolean(data.bot);
+    if ('bot' in data || typeof this.bot !== 'boolean') {
+      /**
+       * Whether or not the user is a bot
+       * @type {boolean}
+       */
+      this.bot = Boolean(data.bot);
+    }
```

Comment:
Needs an `else this.bot = null`

Reply:
That would cause the prop to erroneously become `null` instead of `false` then.
Not overwriting `bot` if we have nothing to assign in an update is the intention with this PR.

Reply:
It wouldn't cause the prop to erroneously become `null`? It's literally what the properties above and below this one do...

Reply:
>It wouldn't cause the prop to erroneously become `null`?

It would, see below.

> It's literally what the properties above and below this one do...

I do not understand that part, if there is an issue with something, please elaborate.

---

The below part:

Adding that `else ...` would cause this behavior:
```js
// Partial user payload
const data = { id: '123' };

// user props
const that = { bot: true, id: '123', username: 'foo', discriminator: '0815' };

if ('bot' in data || typeof that.bot !== 'boolean') {
    that.bot = Boolean(data.bot);
} else that.bot = null;

console.log(that); // { bot: null /* ... other props, unchanged */ } <- should have stayed true
```
Patching a user with partial data, which is what this issue is about, would cause the property to become `null` while it should not be touched at all.

Reply:
Because your code doesn't do the same thing we already to for `username`, `avatar`, `discriminator`, etc...
```js
const data = { id: '123' };

// user props
const that = { bot: true, id: '123', username: 'foo', discriminator: '0815' };

if (Reflect.has(data, 'bot')) {
	that.bot = Boolean(data.bot);
} else if (typeof that.bot !== 'boolean') {
	that.bot = null;
}
```

Reply:
in this case, `bot` will only be in the `data` if it is true, so (as attempted in #4636) `User#bot` will be `null` if the user is not a bot, so we have to assume false if its not present

Reply:
So Discord really doesn't send `bot: false` for normal users... *great* :/
Ignore me then